mjidris/irondb
# React migration

## Objective
This branch handles user authentication and verification through our express backend

## Detailed Description
The react login client will now process login data through an express endpoint that will use passport.js to validate the user's credentials against what's stored in the postgres database. Failed logins will now print error message that's passed down from the backend. Logout button now destroys the express backend session as well.

## Screenshot (if applicable)
![image](https://user-images.githubusercontent.com/48575921/75101776-15688d80-55b0-11ea-8e16-9a1c25476f55.png)
![image](https://user-images.githubusercontent.com/48575921/75101780-244f4000-55b0-11ea-86a8-057d75a9f8ce.png)


## How to Test (if applicable)
**The express frontend is in a partially broken state but it only effects the frontend so there will be a specific testing procedure that I'll cover!**
**John is also working on a fix for a docker issue that we will work around for this testing**

1. Rebuild all your containers.
First remove all existing containers...
`./iron.sh -s`
Next build them all again
`./iron.sh -i`

2. Go to http://localhost:3001/register The frontend may load forever. If it doesn't stop loading simply hit the stop button on your browser and the frontend will draw.
3. Fill out registration and hit Submit. You will receive a 500 error but that's caused by a broken frontend. Your account has been created.
4. Due to a docker issue that we are addressing, the containerized react client can't access the express api yet. In order to work around this, start up a new react client on your host machine.
Navigate to the client directory and run
`npm start`
It should inform you that another client is already running on the default port and ask if you'd like to run another on a different port. Select yes.
5. Navigate to the new react client and attempt to login with incorrect credentials and observe the error message. Open inspector and view the logs for the raw output from express.
6. Login with correct credentials and check inspector logs for raw output.
7. Logout and verify that you are indeed logged out

## Tests (if added)


## Requirements
[Taiga User Story](https://tree.taiga.io/project/cbrobsto-iron-db/us/237)
[Taiga Task](https://tree.taiga.io/project/cbrobsto-iron-db/task/249)
[Taiga Task](https://tree.taiga.io/project/cbrobsto-iron-db/task/240)
[Taiga Task](https://tree.taiga.io/project/cbrobsto-iron-db/task/250)
[Taiga Task](https://tree.taiga.io/project/cbrobsto-iron-db/task/251)
